app/voip/server/api/v1/Rooms.ts

```diff
@@ -0,0 +1,86 @@
+import { Meteor } from 'meteor/meteor';
+import { Match, check } from 'meteor/check';
+import { Random } from 'meteor/random';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { API } from '../../../../api/server';
+import { VoipRoom } from '../../../../models/server/raw';
+import { Voip } from '../../lib/Voip';
+import { OmnichannelSourceType } from '../../../../../definition/IRoom';
+
+API.v1.addRoute('voip/room', {
+	async get() {
+		const defaultCheckParams = {
+			token: String,
+			rid: Match.Maybe(String),
+			agentId: Match.Maybe(String),
+		};
+		check(this.queryParams, defaultCheckParams);
+
+		const { token, rid: roomId, agentId } = this.queryParams;
+		const guest = await Voip.getVisitorByToken(token);
+		if (!guest) {
+			throw new Meteor.Error('invalid-token');
+		}
+
+		let room;
+		if (!roomId) {
+			room = await VoipRoom.findOneOpenByVisitorToken(token, {});
+			if (room) {
+				return API.v1.success({ room, newRoom: false });
+			}
+			let agent;
+			const agentObj = agentId && Voip.findAgent(agentId);
+			if (agentObj) {
+				const { username } = agentObj;
+				agent = { agentId, username };
+			}
+			const rid = Random.id();
+			const roomInfo = {
+				source: {
+					type: OmnichannelSourceType.API,
+				},
+			};
+			room = await Voip.getNewRoom(guest, agent, rid, roomInfo);
+			return API.v1.success(room);
+		}
+		room = await VoipRoom.findOneOpenByRoomIdAndVisitorToken(roomId, token, {});
+		if (!room) {
+			throw new Meteor.Error('invalid-room');
+		}
+		return API.v1.success({ room, newRoom: false });
+	},
+});
+API.v1.addRoute('voip/room.close', {
```

Comment:
I would prefer keeping it in a separate file for the simplicity and readability. 
'voip/room' is slightly different from livechats's 'voip/room' API.  (In theory, based on incoming type parameter, it is possible to write conditional code. But I mentioned in some comment here, we would try to do less modification to livechat code). So we have created a separate endpoint for room creation in context of Voip. Even though the code for closing the room in case of Voip might be the same, it will create a confusion in the mind of reader about where the voip room is getting closed if we reuse the code for livechat's 'room.close' code.